corehq/blobs/tests/test_mixin.py

```diff
@@ -90,6 +90,50 @@ def test_document_blobs(self):
         self.assertEqual(self.obj.blobs[name].content_type, "text/plain")
         self.assertEqual(self.obj.blobs[name].content_length, 7)
 
+    def test_deferred_put_attachment(self):
+        obj = self.make_doc(DeferredPutBlobDocument)
+        name = "test.1"
+        content = StringIO(b"content")
+        obj.deferred_put_attachment(content, name, content_type="text/plain")
+        self.assertEqual(obj.blobs[name].id, None)
+        self.assertEqual(obj.blobs[name].content_type, "text/plain")
+        self.assertEqual(obj.blobs[name].content_length, 7)
+        self.assertFalse(obj.saved)
+
+    def test_put_attachment_overwrites_unsaved_blob(self):
+        obj = self.make_doc(DeferredPutBlobDocument)
+        name = "test.1"
+        obj.deferred_put_attachment(b"<xml />", name, content_type="text/xml")
+        obj.put_attachment(b"new", name, content_type="text/plain")
+        self.assertTrue(obj.blobs[name].id is not None)
+        self.assertEqual(obj.blobs[name].content_type, "text/plain")
+        self.assertEqual(obj.blobs[name].content_length, 3)
+
+    def test_fetch_attachment_returns_unsaved_blob(self):
+        obj = self.make_doc(DeferredPutBlobDocument)
+        name = "test.1"
+        content = b"<xml />"
+        obj.deferred_put_attachment(content, name, content_type="text/xml")
+        self.assertEqual(obj.fetch_attachment(name), content)
+
+    def test_delete_attachment_deletes_unsaved_blob(self):
+        obj = self.make_doc(DeferredPutBlobDocument)
+        name = "test.1"
+        content = b"<xml />"
+        obj.deferred_put_attachment(content, name, content_type="text/xml")
+        self.assertTrue(obj.delete_attachment(name))
+
+    def test_save_persists_unsaved_blob(self):
+        obj = self.make_doc(DeferredPutBlobDocument)
+        name = "test.1"
+        content = StringIO(b"content")
+        obj.deferred_put_attachment(content, name, content_type="text/plain")
+        obj.save()
+        self.assertTrue(obj.saved)
```

Comment:
All couch db is mocked in these tests to keep them fast. This line validates that the base [`obj.save()`](https://github.com/dimagi/commcare-hq/pull/11523/files#diff-8a2de1f87337170d35b54199a1fc7b1eR724) was called, which is assumed to actually save the document in a non-test environment.